service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -178,14 +226,92 @@ private void processResponseHeaders(IoSession session) {
 
-                    boolean upgrade = processHopByHopHeaders(connectSession, acceptSession);
-                    // Add Connection: upgrade to acceptSession
-                    if (upgrade) {
-                        acceptSession.setWriteHeader(HEADER_CONNECTION, HEADER_UPGRADE);
+                    processResponseHeaders(connectSession, acceptSession);
+                }
+
+            }
+        }
+
+        private void processResponseHeaders(HttpSession connectSession, HttpSession acceptSession) {
+
+            Set<String> hopByHopHeaders = getHopByHopHeaders(connectSession);
+            boolean upgrade = connectSession.getReadHeader(HEADER_UPGRADE) != null;
+            if (upgrade) {
+                hopByHopHeaders.remove(HEADER_UPGRADE);
+            }
+
+            // Add processed connect session headers to accept session
+            for (Map.Entry<String, List<String>> e : connectSession.getReadHeaders().entrySet()) {
+                String name = e.getKey();
+                // don't add hop-by-hop response headers
+                if (hopByHopHeaders.contains(name)) {
+                    continue;
+                }
+                for (String value : e.getValue()) {
+                    if (name.equalsIgnoreCase(HEADER_SET_COOKIE)) {
+                        if (rewriteCookieDomain) {
+                            value = processCookieDomain(value, cookieDomainMap);
+                        }
+                        if (rewriteCookiePath) {
+                            value = processCookiePath(value, cookiePathMap);
+                        }
+                        acceptSession.addWriteHeader(name, value);
+                    } else if (name.equalsIgnoreCase(HEADER_LOCATION)) {
+                        if (rewriteLocation) {
+                            value = processLocationHeader(value, locationMap);
+                        }
+                        acceptSession.addWriteHeader(name, value);
+                    } else {
+                        acceptSession.addWriteHeader(name ,value);
                     }
                 }
             }
+
+            // Add Connection: upgrade to acceptSession
+            if (upgrade) {
+                acceptSession.setWriteHeader(HEADER_CONNECTION, HEADER_UPGRADE);
+            }
+
         }
+
+        private String processCookieDomain(String cookie, Map<String, String> cookieDomainMap) {
+            String lowerCookie = cookie.toLowerCase();
+            if (lowerCookie.contains("domain=")) {
+                return cookieDomainMap.entrySet().stream()
+                        .filter(e -> lowerCookie.contains("domain="+e.getKey()))
+                        .findFirst()
+                        .map(e -> {
+                            int index = lowerCookie.indexOf("domain="+e.getKey());
+                            return cookie.substring(0, index+7)+e.getValue()+cookie.substring(index+7+e.getKey().length());
```

Comment:
The 7 letters is from `http://`? Could the scheme be `https://` or something different then 7 letters long?